agent/agent.go

```diff
@@ -563,11 +563,11 @@ func (a *Agent) reloadWatches(cfg *config.RuntimeConfig) error {
 		} else if hasArgs && !ok {
 			return fmt.Errorf("Watch args must be a list of strings")
 		}
-		if hasHandler && hasArgs {
```

Comment:
Also, if the handler type is empty this can set it to "script" before ParseExempt is called. That way the parser layer only knows about two valid types "http" and "script" and doesn't support blank or a default case.